pkg/solana/chainwriter/transform_registry.go

```diff
@@ -12,25 +11,44 @@ import (
 	"github.com/mitchellh/mapstructure"
 
 	idl "github.com/smartcontractkit/chainlink-ccip/chains/solana"
-	"github.com/smartcontractkit/chainlink-ccip/chains/solana/gobindings/ccip_offramp"
 	"github.com/smartcontractkit/chainlink-ccip/chains/solana/gobindings/ccip_router"
 	ccipconsts "github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+
 	"github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
 
 	"github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
+	txmutils "github.com/smartcontractkit/chainlink-solana/pkg/solana/txm/utils"
 )
 
-const MandatoryExecuteAccounts = 14
+type ExtraDataDecoded struct {
+	// ExtraArgsDecoded contain message specific extra args.
+	ExtraArgsDecoded map[string]any
+	// DestExecDataDecoded contain token transfer specific extra args.
+	DestExecDataDecoded []map[string]any
+}
+
+type SVMExecCallArgs struct {
+	ReportContext [2][32]byte                `mapstructure:"ReportContext"`
+	Report        []byte                     `mapstructure:"Report"`
+	Info          ccipocr3.ExecuteReportInfo `mapstructure:"Info"`
+	ExtraData     ExtraDataDecoded           `mapstructure:"ExtraData"`
+	TokenIndexes  []byte                     `mapstructure:"TokenIndexes"`
```

Comment:
In the chainlink repo, I see this defined as:

type SVMExecCallArgs struct {
	ReportContext [2][32]byte                 `mapstructure:"ReportContext"`
	Report        []byte                      `mapstructure:"Report"`
	Info          ccipocr3.ExecuteReportInfo  `mapstructure:"Info"`
	ExtraData     ccipcommon.ExtraDataDecoded `mapstructure:"ExtraData"`
}
https://github.com/smartcontractkit/chainlink/blob/fe50f9203b4d4ce2767e8994992ab645b0649db8/core/capabilities/ccip/ocrimpls/svm_contract_transmitter_factory.go#L35-L40


Why doesn't the TokenIndexes reflect there?

Could you export this type from here or somewhere in chainlink-ccip repo, and make both the chainlink-solana and chainlink repos use the same?

Same for the SVMCommitCallArgs

Reply:
We were planning on importing into `chainlink` from `chainlink-solana` after this goes through. `TokenIndexes` would just be left empty there.